doc/api/http2.md

```diff
@@ -342,6 +342,31 @@ acknowledgement for a sent SETTINGS frame. Will be `true` after calling the
 `http2session.settings()` method. Will be `false` once all sent SETTINGS
 frames have been acknowledged.
 
+#### http2session.ping(callback)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `callback` {Function}
+* Returns: {boolean}
+
+Sends a `PING` frame to the connected HTTP/2 peer. A `callback` function must
+be provided. Only one `PING` may be sent at any given time. The method will
+return `true` if the `PING` was sent, `false` otherwise.
+
+The callback will be invoked with two arguments: a boolean indicating whether
+or not the `PING` was successfully acknowledged, and a `duration` argument that
+reports the number of milliseconds elapsed since the ping was sent and the
+acknowledgement was received.
+
+```js
+session.ping((ack, duration) => {
```

Comment:
If you want to deviate from errback style, it might be a good idea to add a custom `util.promisify` variant (or stick with errback)

Reply:
Yeah, I'm debating back and forth on this one

Reply:
Alternatively, this could accept a `payload` argument and simply send out a `PING` frame. Any received `PING` frames would emit a `ping` event with the `payload` as argument. That avoids state (leaking memory) and allows users to implement promises or callbacks as well as limit or allow parallel PINGs.